#!/usr/bin/env python3

# Set global variables
lab = ["LAB" + str(i+1) for i in range(3)]
replicate = ["LIB" + str(i+1) for i in range(4)]
panel = ["PanelA", "PanelB"]

rule all:
    input:
        "results/multiSample.SNV.classifier",
        "results/multiSample.INDEL.classifier"

pisces_vcf = expand("/home/azureuser/data/{panel}/VCF/unfiltered_VCF_Pisces/{panel}_{lab}_{replicate}.hg19.aligned.duplicates_marked.recalibrated.vcf", panel=panel, lab=lab, replicate=replicate, allow_missing=True)
scalpel_vcf = expand("/home/azureuser/data/{panel}/VCF/unfiltered_VCF_scalpel/{panel}_{lab}_{replicate}_scalpel.vcf", panel=panel, lab=lab, replicate=replicate, allow_missing=True)

# split Pisces to snv and indel
rule split_file:
    input:
        pisces_vcf
    output:
        pisces_indel = expand("/home/azureuser/data/{panel}/VCF/unfiltered_VCF_Pisces/{panel}_{lab}_{replicate}_indels.vcf", panel=panel, lab=lab, replicate=replicate, allow_missing=True),
        pisces_snv = expand("/home/azureuser/data/{panel}/VCF/unfiltered_VCF_Pisces/{panel}_{lab}_{replicate}_snvs.vcf", panel=panel, lab=lab, replicate=replicate, allow_missing=True)
    conda: "/home/azureuser/bin/miniconda3/envs/somaticseq"
    script:
        "splitVcf.py -infile {input.pisces_vcf} -snv {output.pisces_snv} -indel {output.pisces_indel}"

# sort scalpel file based on chromosomes
rule sort_scalpel:
    input:
       scalpel_vcf 
    output:
        scalpel_sort = expand("/home/azureuser/data/{panel}/VCF/unfiltered_VCF_scalpel/{panel}_{lab}_{replicate}_scalpel_sorted.vcf", panel=panel, lab=lab, replicate=replicate, allow_missing=True)
    conda: "/home/azureuser/bin/miniconda3/envs/somaticseq"
    script:
        "vcf-sort -c {input.scalpel_vcf} > {output.scalpel_sort}"

outdir = expand("results/{panel}_{lab}_{replicate}/", panel=panel, lab=lab, replicate=replicate, allow_missing=True)
bam = expand("/home/azureuser/data/{panel}/BAM/Bamout/{panel}_{lab}_{replicate}.hg19.aligned.duplicates_marked.recalibrated-unfiltered.vcf.gz.out.bam", panel=panel, lab=lab, replicate=replicate, allow_missing=True)
mutect2 = expand("/home/azureuser/data/{panel}/VCF/filtered_VCF_M2/{panel}_{lab}_{replicate}.hg19.aligned.duplicates_marked.recalibrated-filtered.vcf.gz", panel=panel, lab=lab, replicate=replicate, allow_missing=True)
vardict = expand("/home/azureuser/data/{panel}/VCF/unfiltered_VCF_VarDict/{panel}_{lab}_{replicate}.vcf", panel=panel, lab=lab, replicate=replicate, allow_missing=True)
strelka = expand("/home/azureuser/data/{panel}/VCF/unfiltered_VCF_Strelka2/{panel}_{lab}_{replicate}_strelka2.vcf.gz", panel=panel, lab=lab, replicate=replicate, allow_missing=True)

# somaticseq parallel
rule somaticseq_parallel:
    input:        
        scalpel_sort = expand("/home/azureuser/data/{panel}/VCF/unfiltered_VCF_scalpel/{panel}_{lab}_{replicate}_scalpel_sorted.vcf", panel=panel, lab=lab, replicate=replicate, allow_missing=True),
        pisces_indel = expand("/home/azureuser/data/{panel}/VCF/unfiltered_VCF_Pisces/{panel}_{lab}_{replicate}_indels.vcf", panel=panel, lab=lab, replicate=replicate, allow_missing=True),
        pisces_snv = expand("/home/azureuser/data/{panel}/VCF/unfiltered_VCF_Pisces/{panel}_{lab}_{replicate}_snvs.vcf", panel=panel, lab=lab, replicate=replicate, allow_missing=True)
    output:
        expand("results/{panel}_{lab}_{replicate}/Consensus.sSNV.vcf", panel=panel, lab=lab, replicate=replicate, allow_missing=True),
        expand("results/{panel}_{lab}_{replicate}/Consensus.sINDEL.vcf", panel=panel, lab=lab, replicate=replicate, allow_missing=True),
        expand("results/{panel}_{lab}_{replicate}/Ensemble.sSNV.tsv", panel=panel, lab=lab, replicate=replicate, allow_missing=True),
        expand("results/{panel}_{lab}_{replicate}/Ensemble.sINDEL.tsv", panel=panel, lab=lab, replicate=replicate, allow_missing=True)
    conda: "/home/azureuser/bin/miniconda3/envs/somaticseq"
    script:
        """
        somaticseq_parallel.py \
          --output-directory {wildcards.outdir} \
          --genome-reference /home/azureuser/data/ref/hg19.fa \
          single \
          --bam-file {wildcards.bam}
          --mutect2-vcf {wildcards.mutect2} \
          --vardict-vcf {wildcards.vardict} \
          --scalpel-vcf {input.scalpel_sort} \
          --strelka-vcf {wildcards.strelka} \
          --arbitrary-snvs {input.pisces_snv} \
          --arbitrary-indels {input.pisces_indel} \
        """
        
# somaticseq classifier
def snv_tsvs(wildcards):
    snvs = expand("results/{panel}_{lab}_{replicate}/Ensemble.sSNV.tsv", panel=panel, lab=lab, replicate=replicate, allow_missing=True)
    print(snvs)
    return snvs

def indel_tsvs(wildcards):
    indels = expand("results/{panel}_{lab}_{replicate}/Ensemble.sINDEL.tsv", panel=panel, lab=lab, replicate=replicate, allow_missing=True)
    print(indels)
    return indels

rule somaticseq_classifier:
    input: 
      snv_tsvs,
      indel_tsvs
    output: 
      "results/multiSample.SNV.classifier",
      "results/multiSample.INDEL.classifier"
    conda: "/home/azureuser/bin/miniconda3/envs/somaticseq"
    script: 
      """
      somatic_xgboost.py train \
        -tsvs {input.snv_tsvs} \
        -out results/multiSample.SNV.classifier \
        -threads 8 -depth 12 -seed 42 -method hist -iter 250 \
        --extra-params scale_pos_weight:0.1 grow_policy:lossguide max_leaves:12

      somatic_xgboost.py train \
        -tsvs {input.indel_tsvs} \
        -out results/multiSample.INDEL.classifier \
        -threads 8 -depth 12 -seed 42 -method hist -iter 250 \
        --extra-params scale_pos_weight:0.1 grow_policy:lossguide max_leaves:12
      """

